pkg/resourcesets/collector.go

```diff
@@ -235,24 +235,22 @@ func (h *ResourceHandler) filterByNameAndLabel(ctx context.Context, dr dynamic.R
 		// no filters for names of the resource, return all objects obtained from the list call
 		return resourceObjectsList.Items, nil
 	}
+
 	// filter out using ResourceNameRegexp
 	if filter.ResourceNameRegexp != "" {
 		logrus.Debugf("Using ResourceNameRegexp [%s] to filter resource names", filter.ResourceNameRegexp)
-		if filter.ResourceNameRegexp == "." {
```

Comment:
This is a different bug unrelated to the one we were trying to fix - HOWEVER, fixing this bug was a happy accident along the way. It helps ensure that our other fixes (which in general make things much less greedy) don't cause other unexpected bugs.